### DISCUSSION TITLE
How can I adjust hovernet to 0.5

### DISCUSSION THREAD

**Discussion by OmarAshkar** [Project-MONAI/MONAI#5922] [Q&A]
Hello,

I am trying to use hovernet implementation in MONAI. I successfuly trained a model but it seems that some objects are being removed cause of 0.5 mpp resolution I have.

the original mpp of hovernet is 0.25, then there is hovernetplus with mpp 0.5. I believe it was a single line of code change: 
https://github.com/TissueImageAnalytics/tiatoolbox/blob/aab2df9c8169ae87c433eedf42d9232f526c5339/tiatoolbox/models/architecture/hovernetplus.py#L293

I cannot find any equivalent for this postprocess in MONAI implementation of hovernet.

Any help is appreciated. 

Thanks!

**Comment by KumoLiu** [comment]
Hi @OmarAshkar, from the code you shared, he proposed a way to calculate the relationship between mpp and the kernel size of sobel, the size of small object, which is currently not added in MONAI implementation. 
https://github.com/TissueImageAnalytics/tiatoolbox/blob/aab2df9c8169ae87c433eedf42d9232f526c5339/tiatoolbox/models/architecture/hovernet.py#L535-L536

But you could achieve the same effect by specifying `min_object_size` and `sobel_kernel_size` these two parameters in `HoVerNetInstanceMapPostProcessing` directly.
https://github.com/Project-MONAI/MONAI/blob/1a018a7b3034a86360d999a6bcc796bad330bba4/monai/apps/pathology/transforms/post/array.py#L652

Hope it can help you, thanks!

**Reply by OmarAshkar** [reply]
@KumoLiu Super helpful! to reiterate, there is no post processing step in MONAI that would remove the small objects anyway, so I can use any mpp, right?

**Reply by KumoLiu** [reply]
Hi @OmarAshkar, any mpp is OK, in fact. I'm not sure what you mean about removing small objects. There are two places using `remove small objects`, IMO it doesn't affect what mpp you use.
https://github.com/Project-MONAI/MONAI/blob/1a018a7b3034a86360d999a6bcc796bad330bba4/monai/apps/pathology/transforms/post/array.py#L693-L695
https://github.com/Project-MONAI/MONAI/blob/1a018a7b3034a86360d999a6bcc796bad330bba4/monai/apps/pathology/transforms/post/array.py#L698-L703

Hope it can help you, thanks!

**Reply by OmarAshkar** [reply]
@KumoLiu I followed your tutorial and my segmentation results does not include small cells. My resolution is 20X. So, I will play around the min_object_size and see if it enhances the result. 

Many thanks!

**Reply by KumoLiu** [reply]
Hi @OmarAshkar, that make sense, maybe you could set `min_object_size=-1`, then it will not remove small objects.
https://github.com/Project-MONAI/MONAI/blob/1a018a7b3034a86360d999a6bcc796bad330bba4/monai/apps/pathology/transforms/post/array.py#L319
Thanks!
